validator/tests/test_consensus/tests.py

```diff
@@ -284,29 +284,29 @@ def test_cancel_block(self):
     # Using chain controller
     def test_check_blocks(self):
         block_ids = [bytes([0x56]), bytes([0x78])]
-        self._mock_block_cache["56"] = "block0"
-        self._mock_block_cache["78"] = "block1"
+        self._mock_block_manager["56"] = "block0"
+        self._mock_block_manager["78"] = "block1"
         self._proxy.check_blocks(block_ids)
 
         with self.assertRaises(UnknownBlock):
             self._proxy.check_blocks([bytes([0x00])])
 
     def test_commit_block(self):
-        self._mock_block_cache["34"] = "a block"
+        self._mock_block_manager["34"] = "a block"
```

Comment:
I am confused by how this works, since ConsensusProxy expects to be able to call `block_manager.get(["34"])` and get back an iterator, but this fails with `TypeError: unhashable type: 'list'` when I try it manually. I think you need to make the mock act more like the real thing.

Reply:
I changed to have a MockBlockManager class that implements `get(block_ids)` but also __setitem__ so that this still works. The reason  why that is nice is so you can specify the header_signature separate from the "block". In the case of the one you called out, the string "a block" doesn't have an attribute `header_signature` and it save me some work in constructing blocks that do.